apiServer.js

```diff
@@ -10,13 +11,47 @@ const app = require('./ZelBack/src/lib/server');
 const log = require('./ZelBack/src/lib/log');
 const serviceManager = require('./ZelBack/src/services/serviceManager');
 const upnpService = require('./ZelBack/src/services/upnpService');
-const userconfig = require('./config/userconfig');
 
 const cmdAsync = util.promisify(nodecmd.get);
+const md5 = require('md5');
 
 const apiPort = userconfig.initial.apiport || config.server.apiport;
 const apiPortHttps = apiPort + 1;
 
+const { watch } = require("fs/promises");
+let md5Previous = null;
+let initialMd5 = null;
+
+async function configReload() {
+  try {
+    const watcher = watch(path.join(__dirname, '/config'));
+    for await (const event of watcher) {
+      if ( event.eventType === 'change' && event.filename === 'userconfig.js') {
+        const md5Current = md5(fs.readFileSync(path.join(__dirname, '/config/userconfig.js')));
+        if (md5Current === md5Previous) {
+          return;
+        }
+        md5Previous = md5Current
+        if( initialMd5 === null) {
+          initialMd5 = md5Current
+        }
+        if ( initialMd5 !== md5Current ) {
+          initialMd5 = null
+          console.log(`Config file changed, reloading ${event.filename}...`);
+        }
+        delete require.cache[require.resolve('./config/userconfig')];
+        userconfig = require('./config/userconfig');
+      }
+    }
+  } catch (error) {
+      console.error(`Error watching files: ${error}`);
+  }
+}
+
+setInterval(async () => {
+  configReload();
+}, 2 * 1000);
```

Comment:
lets make it slower, 60 * 1000
once per minute is more than enough.

Reply:
i dont agree it should work fast ( hot reload ) same way as working nodemon

Reply:
or set it global as 30s  and import function and use it when any cmd with config is use

Reply:
ok let it be every 2 seconds